lib/start-server.js

```diff
@@ -37,23 +37,29 @@ const program = require('commander');
 
 program.option('--port <number>', 'Specify port number').parse(process.argv);
 
-const port = parseInt(program.port, 10) || 3000;
-
-tcpPortUsed
-  .check(port, 'localhost')
-  .then(function(inUse) {
-    if (inUse) {
-      console.error(chalk.red('Port ' + port + ' is in use'));
-      process.exit(1);
-    } else {
-      console.log('Starting Docusaurus server on port ' + port + '...');
-      // start local server on specified port
-      const server = require('./server/server.js');
-      server(port);
-    }
-  })
-  .catch(function(ex) {
-    setTimeout(function() {
-      throw ex;
-    }, 0);
-  });
+var port = process.env.PORT || 3000;
+checkPort();
+
+function checkPort() {
+  tcpPortUsed
+    .check(port, 'localhost')
+    .then(function(inUse) {
+      if (inUse) {
+        console.error(chalk.red('Port ' + port + ' is in use'));
+        // Try again but with port + 1
+        port += 1;
+        checkPort();
+        // process.exit(1);
```

Comment:
This line should be entirely removed.

Reply:
Hm, do you mean that those 4 lines should all be removed?

Reply:
No, just the line you commented out